tests/Controller/ModelCollectorTest.php

```diff
@@ -206,6 +214,328 @@ public function testCollectSiblingsOf()
+     */
+    public function testSearchParentOfInHierarchicalByInverseFilter(
+        ?ModelInterface $expected,
+        ModelInterface $model
+    ): void {
+        $definition      = $this->mockDefinitionContainer();
+        $basicDefinition = $this->mockBasicDefinition();
+        $basicDefinition->method('getDataProvider')->willReturn('node');
+        $basicDefinition->method('getRootDataProvider')->willReturn('node');
+        $basicDefinition->method('getParentDataProvider')->willReturn('parent');
+        $basicDefinition->method('getMode')->willReturn(BasicDefinitionInterface::MODE_HIERARCHICAL);
+        $definition->method('getBasicDefinition')->willReturn($basicDefinition);
+
+        $relationships = $this->mockRelationshipDefinition();
+        $relationships->method('getChildConditions')->willReturn(
+            [
+                $this->createParentChildCondition('parent', 'node'),
+                $this->createParentChildCondition('node', 'node'),
+            ]
+        );
+
+        $rootCondition = $this->getMockForAbstractClass(RootConditionInterface::class);
+        $relationships->method('getRootCondition')->willReturn($rootCondition);
+
+        $definition->method('getModelRelationshipDefinition')->willReturn($relationships);
+
+        $environment = $this->getMockForAbstractClass(EnvironmentInterface::class);
+        $environment->method('getDataDefinition')->willReturn($definition);
+
+        $config = $this->getMockForAbstractClass(ConfigInterface::class);
+        $config->method('setFilter')->willReturn($config);
+
+        $parentProvider = $this->getMockForAbstractClass(DataProviderInterface::class);
+        $parentProvider->method('getEmptyConfig')->willReturn($config);
+        $parentProvider
+            ->expects($this->once())
+            ->method('fetch')
+            ->willReturn(null);
+
+        $provider = $this->getMockForAbstractClass(DataProviderInterface::class);
+        $provider->method('getEmptyConfig')->willReturn($config);
+        $provider
+            ->expects($this->once())
+            ->method('fetch')->willReturn($expected);
+
+        $provider
+            ->expects(null === $expected ? $this->once() : $this->never())
+            ->method('fetchAll')
+            ->willReturn(new DefaultCollection());
+
+        $environment->method('getDataProvider')->willReturnCallback(
+            function (string $name) use ($provider, $parentProvider) {
+                switch ($name) {
+                    case 'node':
+                        return $provider;
+                    case 'parent':
+                        return $parentProvider;
+
+                    default:
+                        throw new \RuntimeException();
+                }
+            }
+        );
+
+        $collector = new ModelCollector($environment);
+        $this->assertSame($expected, $collector->searchParentOf($model));
```

Comment:
Please call static method via `self::assertSame()`

Reply:
Fixed in 7826ab36bea4d62ae0edbd3b8c1e06bdbe44bc44